File: src/leetcode/count-equal-and-divisible-pairs-in-an-array/count-equal-and-divisible-pairs-in-an-array.js
/**
 * Leetcode #2176
 * Difficulty: Easy
 * count-equal-and-divisible-pairs-in-an-array
 */
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
export const countPairs = (nums, k) => {
  let answer = 0;

  for (let i = 0; i < nums.length; i += 1) {
    for (let j = i + 1; j < nums.length; j += 1) {
      if (0 <= i && i <= j && nums[i] === nums[j] && (i * j) % k === 0) {
        answer += 1;
      }
    }
  }

  return answer;
};

File: src/leetcode/count-equal-and-divisible-pairs-in-an-array/count-equal-and-divisible-pairs-in-an-array.rs
impl Solution {
    pub fn count_pairs(mut nums: Vec<i32>, k: i32) -> i32 {
        let mut answer = 0;
        let len = nums.len();

        for i in 0..len - 1 {
            for j in i + 1..nums.len() {
                if 0 <= i && i <= j && nums[i] == nums[j] && i * j % k as usize == 0 {
                    answer += 1;
                }
            }
        }

        answer
    }
}
